node/hyperbahn/handler.js

```diff
@@ -193,27 +196,41 @@ HyperbahnHandler.prototype.sendRelayAdvertise =
 function sendRelayAdvertise(hostPort, services, callback) {
     var self = this;
 
-    self.tchannelJSON.send(self.channel.request({
-        host: hostPort,
-        serviceName: 'hyperbahn'
-    }), 'relay-ad', null, {
-        services: services
-    }, onRelayAdvertise);
+    var attempts = 0;
```

Comment:
prefer this to be passed as function arg... could even then hoist it to a method, so we have:
```
HH#sendRelayAd(h, s, cb) {
  self._tryRelayAd(0, h, s, cb);
};

HH#_tryRelayAd(attempt, h, s, cb) {
  ...
  if (maybe) self._tryRelayAd(attemp + 1, h, s, cb);
  else cb(err, res);
};
```